lib/src/pubsub_impl.dart

```diff
@@ -477,8 +477,10 @@ class _TopicPageImpl implements Page<Topic> {
   bool get isLast => _nextPageToken == null;
 
   @override
-  Future<Page<Topic>> next({int? pageSize}) {
-    if (isLast) return Future.value(null);
+  Future<Page<Topic>> next({int? pageSize}) async {
+    if (isLast) {
+      throw StateError('Page.next() cannot be called when Page.isLast == true');
```

Comment:
Could this be DRY'd up? This code is repeated a lot

Reply:
Moved it to a new file `common_utils.dart` not sure it's prettier, but I guess it works and we don't repeat the message :D